tt_metal/fabric/control_plane.cpp

```diff
@@ -1012,4 +1013,26 @@ void ControlPlane::set_routing_mode(uint16_t mode) {
 
 uint16_t ControlPlane::get_routing_mode() const { return this->routing_mode_; }
 
+void ControlPlane::initialize_fabric_context() {
+    TT_FATAL(this->fabric_context_ == nullptr, "Trying to re-initialize fabric context");
+    auto fabric_config = tt::tt_metal::MetalContext::instance().get_cluster().get_fabric_config();
+    this->fabric_context_ = std::make_unique<FabricContext>(fabric_config);
+}
+
+FabricContext* ControlPlane::get_fabric_context() const {
+    TT_FATAL(this->fabric_context_ != nullptr, "Trying to get un-initialized fabric context");
+    return this->fabric_context_.get();
+}
+
+void ControlPlane::clear_fabric_context() { this->fabric_context_.reset(nullptr); }
+
+ControlPlane::~ControlPlane() {
+    this->routing_table_generator_.reset(nullptr);
+    this->logical_mesh_chip_id_to_physical_chip_id_mapping_.clear();
+    this->router_port_directions_to_physical_eth_chan_map_.clear();
+    this->intra_mesh_routing_tables_.clear();
+    this->inter_mesh_routing_tables_.clear();
+    this->fabric_context_.reset(nullptr);
+}
```

Comment:
```suggestion
ControlPlane::~ControlPlane() = default;
}
```

The difference between what was there and caused the error message vs. what @omilyutin-tt is suggesting is _when_ the compiler attempts to synthesize the d'tor.  When it's in the header (where it was) then it will be synthesized in every TU.  And with the fwd decl it's unable to do so in other TUs.
But by having it here, all other TUs assume "the body will be provided by some other object file".  And in here it has all the necessary bits to build it.  Best of both worlds.

Reply:
Oh yes, haven't noticed this thread. Thanks @afuller-TT for the commit suggestion!